src/mgr/DaemonServer.cc

```diff
@@ -1643,20 +1643,23 @@ bool DaemonServer::_handle_command(
 	    continue;
 	  }
 	  touched_pgs++;
-	  if (!(q.second.state & PG_STATE_ACTIVE) ||
-	      (q.second.state & PG_STATE_DEGRADED)) {
-	    ++dangerous_pgs;
-	    continue;
-	  }
+
 	  const pg_pool_t *pi = osdmap.get_pg_pool(q.first.pool());
 	  if (!pi) {
 	    ++dangerous_pgs; // pool is creating or deleting
-	  } else {
-	    if (pg_acting.size() < pi->min_size) {
-	      ++dangerous_pgs;
-	    }
+            continue;
 	  }
-	}
+
+	  if (!(q.second.state & PG_STATE_ACTIVE) ||
+              ((pi->type != pg_pool_t::TYPE_REPLICATED) &&
+               (q.second.state & PG_STATE_DEGRADED))) {
+	    ++dangerous_pgs;
+	    continue;
+	  }
```

Comment:
@liewegas I made change 9750061d5d4236aaba156d60790e0b8bcd7cfb64 to add degraded handling but didn't remote the PG_STATE_DEGRADED check here.  So shouldn't this just read?

```
if (!(q.second.state & PG_STATE_ACTIVE)) {
	    ++dangerous_pgs;
 	    continue;
 	  }
```

Reply:
yep!